etcdserver/api/membership/member.go

```diff
@@ -51,6 +51,12 @@ type Member struct {
 	Attributes
 }
 
+// ConfigChangeContext represents a context for confChange.
+type ConfigChangeContext struct {
+	Member
+	IsPromote bool `json:"isPromote"`
```

Comment:
Do we want to make this json `omitempty`? Might be helpful on making it compatible with existing tests. I am not sure about this, just pointing out.

Reply:
can we enable the full etcd ci test in this repo, It works fine in several relevant tests which I start manually on my working machine, not quite sure about whether  the other test will pass.

Reply:
I am not familar with how to enable it. I can take a look tomorrow. For now you can run the same tests locally. Please take a look at the test script under main folder.

Reply:
When I submit changes, I always do unit test and fmt test, sometimes do integration test.

Reply:
I think it is ok now. It is compatible with existing tests. I have it tested,

Reply:
I have done the same as u do